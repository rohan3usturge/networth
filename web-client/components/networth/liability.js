import { NUMBER_UTILS } from "../utils";
import { Amount } from "./amount";

const Liability = ({ liability, onChange, disabled, currency }) => {
  const handleChange = (changedAmount) => {
    const changed = { ...liability };
    changed.lineItem.amount = changedAmount;
    if (onChange) {
      onChange(changed);
    }
  };

  return (
    <tr>
      <td>{liability.lineItem.name}</td>
      <td>
        {NUMBER_UTILS.convertNumToMoney(liability.monthlyPayment, currency)}
      </td>
      <td>
        <Amount
          disabled={disabled}
          defaultValue={liability.lineItem.amount}
          currency={currency}
          className="form-control"
          onChange={handleChange}
        />
      </td>
    </tr>
  );
};

export { Liability };
