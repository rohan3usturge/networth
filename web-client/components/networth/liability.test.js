import { shallow } from "enzyme";
import { Liability } from "./liability";

jest.mock("next/router", () => {
  return {
    useRouter: () => ({ locale: "en-ca" }),
  };
});

describe("Liability", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(
      <Liability
        liability={{ meta: {}, lineItem: {} }}
        currency="CAD"
        disabled={true}
        onChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
