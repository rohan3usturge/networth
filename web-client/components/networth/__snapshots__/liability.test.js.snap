// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Liability should render properly and match snapshot 1`] = `
<tr>
  <td />
  <td />
  <td>
    <Amount
      className="form-control"
      currency="CAD"
      disabled={true}
      onChange={[Function]}
    />
  </td>
</tr>
`;
